### PR TITLE
feat: add Runner.autobisect()

### PR DESCRIPTION
**Pull request by ckerr**
Add `Runner.autobisect()` + full coverage testing.

Found a couple of edge cases, e.g. we want to make sure that the 'good' version is actually good and 'bad' version is actually bad -- the user might be wrong / the test doesn't work everywhere / etc. So we do now check that the 'good' version and 'bad' version didn't return the same RunResult.

Getting into the home stretch now; GUI + CLI are up next.

### CODE REVIEW

File: src/renderer/runner.ts [hunk 1]
```diff
@@ -21,6 +27,12 @@ export enum ForgeCommands {
   MAKE = 'make',
 }
 
+const resultString: Record<RunResult, string> = Object.seal({
```
Comment (erickzhao) [inline src/renderer/runner.ts:30]: nit: Is there a preference for `Object.seal` over `Object.freeze` here?

**Review by erickzhao — APPROVED** [review]
LGTM!

File: src/renderer/runner.ts (hunk 1 above)
Comment (ckerr) [reply src/renderer/runner.ts:30]: Good nit. Using `Object.freeze()` here is better.

Comment (ckerr) [reply src/renderer/runner.ts]: Changed in b72dffc2183adaa8cf9b1e323219a7505f2cb6c9
